### DISCUSSION TITLE
Loading screen

### DISCUSSION THREAD

**Discussion by CHolitas** [swup/swup#391] [Help] [answered]
Question, how do I do a loading screen appear and then the content?

**Comment by AugustMiller** [comment, marked as answer]
Take a look at the [basic example](https://swup.js.org/getting-started/example).

If you want something to appear before your page is "loaded," you can apply the `is-animating` class to the HTML document coming from the server. Swup will remove it when the page after initializing itself!

The same class will be added again on every subsequent transition, as well—so if you need something visible only on the initial page load, you might be better served by applying a different class (`.my-first-load`) and removing it manually when you're ready to display the page:

```js
window.addEventListener('DOMContentLoaded', (event) => {
    window.setTimeout(() => {
        document.documentElement.classList.remove('my-first-load');
    }, 3000);
});

**Comment by simarsudo** [comment]
Answer for someone in future.

There is a loading plugin you can use [link](https://swup.js.org/plugins/progress-plugin)
